### PR TITLE
Add support for zippered

### PR DESCRIPTION
**Pull request by chrisburr**
Opening as a draft to debug building as I'm seeing weird error messages when building locally.

Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/patches/1-support-zippered.patch [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+ src/llvm/projects/libtapi/include/tapi/Core/Platform.h    | 2 ++
+ .../projects/libtapi/include/tapi/LinkerInterfaceFile.h   | 2 ++
+ src/llvm/projects/libtapi/lib/Core/Platform.cpp           | 8 ++++++++
+ src/llvm/projects/libtapi/lib/Core/YAML.cpp               | 3 +++
+ 4 files changed, 15 insertions(+)
+
+diff --git a/src/llvm/projects/libtapi/include/tapi/Core/Platform.h b/src/llvm/projects/libtapi/include/tapi/Core/Platform.h
+index afe19df8a..3293d48a7 100644
+--- a/src/llvm/projects/libtapi/include/tapi/Core/Platform.h
++++ b/src/llvm/projects/libtapi/include/tapi/Core/Platform.h
```
Comment (isuruf) [inline recipe/patches/1-support-zippered.patch:16]: These paths are wrong. You need to remove `src/llvm/projects/libtapi`.

Comment (chrisburr) [reply recipe/patches/1-support-zippered.patch:16]: I think conda-build (or maybe `patch`) is smart enough to fix them automatically:
```
Applying patch: '/Users/christopherburr/Development/conda-forge/tapi-feedstock/recipe/patches/1-support-zippered.patch'
Trying to apply patch as-is
INFO:conda_build.source:Trying to apply patch as-is
INFO conda_build.source:apply_patch(596): Trying to apply patch as-is
patching file include/tapi/Core/Platform.h
patching file include/tapi/LinkerInterfaceFile.h
patching file lib/Core/Platform.cpp
patching file lib/Core/YAML.cpp
Applying patch: '/Users/christopherburr/Development/conda-forge/tapi-feedstock/recipe/patches/2-support-zippered.patch'
patching file tools/libtapi/LinkerInterfaceFile.cpp
```

**Comment by chrisburr** [comment]
@isuruf I found the issue I have building locally. Is there a reason why `clang`/`clang++` are being used instead of `$CC`/`$CXX`?

```bash
cmake \
    -G Ninja \
    -C $SRC_DIR/tapi/cmake/caches/apple-tapi.cmake \
    -DCMAKE_INSTALL_PREFIX=$PREFIX \
    -DCMAKE_C_COMPILER=clang \
    -DCMAKE_CXX_COMPILER=clang++ \
    -DCMAKE_ASM_COMPILER=clang \
    $SRC_DIR/llvm
```

**Comment by isuruf** [comment]
Can't remember. We can try without it.

**Comment by chrisburr** [comment]
I’ve ran out of ideas/time for now. The master branch no longer builds, even if I force the dependencies to match with the last successful build or run it in azure (#2). Any ideas?

I’ll try to look into it more tomorrow.

**Comment by isuruf** [comment]
Thanks.

**Pull request merged**